html/modules/custom/reliefweb_subscriptions/src/Controller/SubscriptionAdminForm.php

```diff
@@ -0,0 +1,152 @@
+<?php
+
+namespace Drupal\reliefweb_subscriptions\Controller;
+
+use Drupal\Core\Controller\ControllerBase;
+use Drupal\Core\Database\Connection;
+use Drupal\Core\Datetime\DateFormatter;
+use Drupal\Core\Link;
+use Drupal\Core\Url;
+use Drupal\reliefweb_subscriptions\CronExpressionParser;
+use Drupal\reliefweb_subscriptions\ReliefwebSubscriptionsMailer;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Manage subscription for user.
+ */
+class SubscriptionAdminForm extends ControllerBase {
+
+  /**
+   * The database connection.
+   *
+   * @var \Drupal\Core\Database\Connection
+   */
+  protected $database;
+
+  /**
+   * The date formatter.
+   *
+   * @var \Drupal\Core\Datetime\DateFormatter
+   */
+  protected $dateFormatter;
+
+  /**
+   * The actual mailer.
+   *
+   * @var \Drupal\reliefweb_subscriptions\ReliefwebSubscriptionsMailer
+   */
+  protected $mailer;
+
+  /**
+   * {@inheritdoc}
+   */
+  public function __construct(Connection $database, DateFormatter $date_formatter, ReliefwebSubscriptionsMailer $mailer) {
+    $this->database = $database;
+    $this->dateFormatter = $date_formatter;
+    $this->mailer = $mailer;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container) {
+    return new static(
+      $container->get('database'),
+      $container->get('date.formatter'),
+      $container->get('reliefweb_subscriptions.mailer'),
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function adminOverview() {
+    $subscriptions = reliefweb_subscriptions_subscriptions();
+
+    // Get the sending log information for the subscriptions.
+    $query = $this->database->select('reliefweb_subscriptions_logs', 'l');
+    $query->fields('l', ['sid', 'last', 'next']);
+    $result = $query->execute();
+    $logs = !empty($result) ? $result->fetchAllAssoc('sid') : [];
```

Comment:
`fetchAllAssoc` returns a list of objects but line 93/94 below expect arrays. This needs ` \PDO::FETCH_ASSOC` as second argument.

```suggestion
    $logs = !empty($result) ? $result->fetchAllAssoc('sid', \PDO::FETCH_ASSOC) : [];
```